llvm/lib/CodeGen/AsmPrinter/DwarfUnit.cpp

```diff
@@ -589,7 +589,8 @@ DIE *DwarfUnit::createTypeDIE(const DIScope *Context, DIE &ContextDIE,
       // Skip updating the accelerator tables since this is not the full type.
       if (MDString *TypeId = CTy->getRawIdentifier()) {
         addGlobalType(Ty, TyDIE, Context);
-        DD->addDwarfTypeUnitType(getCU(), TypeId->getString(), TyDIE, CTy);
+        if (DD->addDwarfTypeUnitType(getCU(), TypeId->getString(), TyDIE, CTy))
+          updateAcceleratorTables(Context, Ty, TyDIE);
```

Comment:
Why not add this to the addDwarfTypeUnitType function, rather than returning then adding here?

Reply:
I was hesitant to make updateAcceleratorTables public, and thought doing it original way would be "smaller scope".
Updated.